client/js/socket.js

```diff
@@ -3,53 +3,46 @@
 const $ = require("jquery");
 const io = require("socket.io-client");
 const path = window.location.pathname + "socket.io/";
+const status = $("#loading-page-message, #connection-error");
 
 const socket = io({
 	transports: $(document.body).data("transports"),
 	path: path,
 	autoConnect: false,
-	reconnection: false
+	reconnection: !$(document.body).hasClass("public")
```

Comment:
I get that we wouldn't want page refresh on public mode, but is this completely disabling client reconnection on public mode? If so, why?

Reply:
Because reconnecting in public mode is effectively useless, it will destroy all the state.

Reply:
So if client loses connection in public mode (in contrast with willingly closing the tab or logging out), it's gone forever? I get that we're doing that currently, but with automatic reconnection possible, that's bit harsh lol

Reply:
If we start adding logic on the server to not destroy the connection as soon as socket closes the connection, we most likely will be opening ourselves to abuse.

Reply:
Agreed, though the more I think about it, the more I see public mode as short-lived sessions with no authentication/synchronization across devices possible. Absolutely not something for this PR, but I'd be curious to see experiments around this to see how it looks and plays.